File: WoofAndroid/data/datasource/src/main/kotlin/com/perrystreet/woof/datasource/dogs/fixtures/Neighborhoods.kt
package com.perrystreet.woof.datasource.dogs.fixtures

internal data class Neighborhood(
    val name: String,
    val climate: Climate,
)

internal object Neighborhoods {
    val all: List<Neighborhood> = listOf(
        Neighborhood("Södermalm, Stockholm", Climate.Cold),
        Neighborhood("Nørrebro, Copenhagen", Climate.Cold),
        Neighborhood("Plateau, Montreal", Climate.Cold),
        Neighborhood("Kensington Market, Toronto", Climate.Cold),
        Neighborhood("Wicker Park, Chicago", Climate.Cold),
        Neighborhood("Williamsburg, Brooklyn", Climate.Temperate),
        Neighborhood("Mission District, San Francisco", Climate.Temperate),
        Neighborhood("Capitol Hill, Seattle", Climate.Temperate),
        Neighborhood("Shoreditch, London", Climate.Temperate),
        Neighborhood("Notting Hill, London", Climate.Temperate),
        Neighborhood("Le Marais, Paris", Climate.Temperate),
        Neighborhood("Kreuzberg, Berlin", Climate.Temperate),
        Neighborhood("Prenzlauer Berg, Berlin", Climate.Temperate),
        Neighborhood("Schöneberg, Berlin", Climate.Temperate),
        Neighborhood("Jordaan, Amsterdam", Climate.Temperate),
        Neighborhood("Shimokitazawa, Tokyo", Climate.Temperate),
        Neighborhood("Fitzroy, Melbourne", Climate.Temperate),
        Neighborhood("Ponsonby, Auckland", Climate.Temperate),
        Neighborhood("Kolonaki, Athens", Climate.Warm),
        Neighborhood("Exarcheia, Athens", Climate.Warm),
        Neighborhood("Glyfada, Athens", Climate.Warm),
        Neighborhood("Silver Lake, Los Angeles", Climate.Warm),
        Neighborhood("Trastevere, Rome", Climate.Warm),
        Neighborhood("Gràcia, Barcelona", Climate.Warm),
        Neighborhood("Malasaña, Madrid", Climate.Warm),
        Neighborhood("Chueca, Madrid", Climate.Warm),
        Neighborhood("Alfama, Lisbon", Climate.Warm),
        Neighborhood("Palermo, Buenos Aires", Climate.Warm),
        Neighborhood("Vila Madalena, São Paulo", Climate.Warm),
        Neighborhood("Surry Hills, Sydney", Climate.Warm),
        Neighborhood("Zona Rosa, Mexico City", Climate.Warm),
    )

    fun matching(climates: Set<Climate>): List<Neighborhood> = all.filter { it.climate in climates }
}

File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSource/Dogs/Fixtures/Neighborhoods.swift
import Foundation

struct Neighborhood {
    let name: String
    let climate: Climate
}

enum Neighborhoods {
    static let all: [Neighborhood] = [
        Neighborhood(name: "Södermalm, Stockholm", climate: .cold),
        Neighborhood(name: "Nørrebro, Copenhagen", climate: .cold),
        Neighborhood(name: "Plateau, Montreal", climate: .cold),
        Neighborhood(name: "Kensington Market, Toronto", climate: .cold),
        Neighborhood(name: "Wicker Park, Chicago", climate: .cold),
        Neighborhood(name: "Williamsburg, Brooklyn", climate: .temperate),
        Neighborhood(name: "Mission District, San Francisco", climate: .temperate),
        Neighborhood(name: "Capitol Hill, Seattle", climate: .temperate),
        Neighborhood(name: "Shoreditch, London", climate: .temperate),
        Neighborhood(name: "Notting Hill, London", climate: .temperate),
        Neighborhood(name: "Le Marais, Paris", climate: .temperate),
        Neighborhood(name: "Kreuzberg, Berlin", climate: .temperate),
        Neighborhood(name: "Prenzlauer Berg, Berlin", climate: .temperate),
        Neighborhood(name: "Schöneberg, Berlin", climate: .temperate),
        Neighborhood(name: "Jordaan, Amsterdam", climate: .temperate),
        Neighborhood(name: "Shimokitazawa, Tokyo", climate: .temperate),
        Neighborhood(name: "Fitzroy, Melbourne", climate: .temperate),
        Neighborhood(name: "Ponsonby, Auckland", climate: .temperate),
        Neighborhood(name: "Kolonaki, Athens", climate: .warm),
        Neighborhood(name: "Exarcheia, Athens", climate: .warm),
        Neighborhood(name: "Glyfada, Athens", climate: .warm),
        Neighborhood(name: "Silver Lake, Los Angeles", climate: .warm),
        Neighborhood(name: "Trastevere, Rome", climate: .warm),
        Neighborhood(name: "Gràcia, Barcelona", climate: .warm),
        Neighborhood(name: "Malasaña, Madrid", climate: .warm),
        Neighborhood(name: "Chueca, Madrid", climate: .warm),
        Neighborhood(name: "Alfama, Lisbon", climate: .warm),
        Neighborhood(name: "Palermo, Buenos Aires", climate: .warm),
        Neighborhood(name: "Vila Madalena, São Paulo", climate: .warm),
        Neighborhood(name: "Surry Hills, Sydney", climate: .warm),
        Neighborhood(name: "Zona Rosa, Mexico City", climate: .warm),
    ]

    static func matching(_ climates: Set<Climate>) -> [Neighborhood] {
        all.filter { climates.contains($0.climate) }
    }
}
